drivers/overlay/composefs.go

```diff
@@ -154,8 +154,38 @@ func mountComposefsBlob(dataDir, mountPoint string) error {
 		mountOpts += "noacl"
 	}
 
-	if err := unix.Mount(loop.Name(), mountPoint, "erofs", unix.MS_RDONLY, mountOpts); err != nil {
-		return fmt.Errorf("failed to mount erofs image at %q: %w", mountPoint, err)
+	fsfd, err := unix.Fsopen("erofs", 0)
```

Comment:
How old is the new mount api? Don't we want to fall back to mount on ENOSYS?

Reply:
I didn't care about the fallback as old kernels do not support anyway the features we need in overlay (and possibly erofs itself)